src/version_edit.h

```diff
@@ -42,5 +45,52 @@ class VersionEdit {
   std::vector<std::pair<uint64_t, SequenceNumber>> deleted_files_;
 };
 
+class EditCollector {
+ public:
+  Status AddEdit(VersionEdit* edit) {
+    auto cf_id = edit->column_family_id_;
+    auto& collector = column_families_[cf_id];
+
+    Status s;
+    for (auto& file : edit->added_files_) {
+      s = collector.AddFile(file);
+      if (!s.ok()) return s;
+    }
+    for (auto& file : edit->deleted_files_) {
+      collector.DeleteFile(file.first, file.second);
+      if (!s.ok()) return s;
+    }
+    return Status::OK();
+  }
+
+ private:
+  friend class VersionSet;
+
+  class Collector {
+   public:
+    Collector() {}
+
+    Status AddFile(const std::shared_ptr<BlobFileMeta>& file) {
+      auto number = file->file_number();
+      if (added_files_.find(number) != added_files_.end()) {
+        return Status::Corruption("blob file has been added before");
+      }
+      added_files_.emplace(number, file);
+      return Status::OK();
+    }
+
+    Status DeleteFile(uint64_t number, SequenceNumber obsolete_sequence) {
+      if (!added_files_.erase(number)) {
```

Comment:
let's collect the set of file added and deleted, and sanity check the file don't appear later on.